src/mozanalysis/historical_targets/__init__.py

```diff
@@ -0,0 +1,109 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+from typing import Optional
+import datetime
+
+import attr
+
+@attr.s(frozen=True, slots=True)
+class TargetDataSource:
```

Comment:
Is it necessary to copy all of this code over from [metrics.DataSource](https://github.com/mozilla/mozanalysis/blob/main/src/mozanalysis/metrics/__init__.py#L12), or can we inherit / reuse some of it? It seems like a lot of code to have duplicated.

Reply:
There are some slight differences between the metrics DataSource and here (similar to differences between SegmentDataSource and DataSource). The closest analog would be Segments, but I defined a new class for Targets following that pattern

Reply:
ah okay - that makes sense.

Reply:
Okay, I've moved the query building from the Target and TargetDataSource to the SegmentDataSource class, so the HistoricalTarget will now take Segment objects